woocommerce/woocommerce
# Product Collection: add price sorting options

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #49527.

This PR:
- adds the following sorting options by price to the Product Collection block:
	- Price, high to low
	- Price, low to high
- ensures the new sorting option works both in the editor and on the frontend.
- Fix the issue with the `update_rest_query_in_editor` method that uses the wrong `orderBy` key, resulting in the PC block not processing the selected order. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a new post.
2. Add the Product Collection to the post. Select 'create your own'.
3. Open the setting sidebar,
4. See the Order by dropdown containing two new options: 'Price, high to low' and 'Price, low to high'.
5. Choose each of these options, and see the editor updated with the right order.
6. Save and see if the order is correct on the front end.
7. Also test other order, see those options working as expected.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
